File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    x_s = board[0]//2
    y_s = board[1]//2
    x, y = 0, 0
    
    for i in keyinput:
        if i=="left":
            if(x<=-x_s):
                x = -x_s
            else:
                x -= 1
        elif i=="right":
            if(x>=x_s):
                x = x_s
            else:
                x+=1
        elif i=="up":
            if(y>=y_s):
                y = y_s
            else:
                y+=1
        elif i=="down":
            if(y<=-y_s):
                y = -y_s
            else:
                y-=1
    answer = [x, y]
    return answer

File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    var answer = [0,0];
    var x = Math.floor(board[0]/2);
    var y = Math.floor(board[1]/2);
    for (let a of keyinput){
        if(a==='left'){
            if((answer[0]-1)<-x){
                continue;
            }
            else{
                answer[0]=answer[0]-1;
            }
        }
        else if (a==='right'){
            if((answer[0]+1)>x){
                continue;
            }
            else{
                answer[0]=answer[0]+1;
            }
        }
        else if (a==='up'){
            if((answer[1]+1)>y){
                continue;
            }
            else{
                answer[1]=answer[1]+1;
            }
        }
        else{
            if((answer[1]-1)<-y){
                continue;
            }
            else{
                answer[1]=answer[1]-1;
            }
        }
    }
    return answer;
}